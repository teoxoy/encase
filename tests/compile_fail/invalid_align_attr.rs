use encase::ShaderType;

fn main() {}

#[derive(ShaderType)]
struct Test {
    #[shader_align]
    a: u32,
    #[shader_align()]
    b: u32,
    #[shader_align(invalid)]
    c: u32,
    #[shader_align(3)]
    d: u32,
}
