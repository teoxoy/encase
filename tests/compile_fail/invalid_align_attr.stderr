error: expected attribute arguments in parentheses: `shader_align(...)`
 --> tests/compile_fail/invalid_align_attr.rs:7:7
  |
7 |     #[shader_align]
  |       ^^^^^^^^^^^^

error: expected a power of 2 u32 literal
 --> tests/compile_fail/invalid_align_attr.rs:9:20
  |
9 |     #[shader_align()]
  |                    ^

error: expected a power of 2 u32 literal
  --> tests/compile_fail/invalid_align_attr.rs:11:20
   |
11 |     #[shader_align(invalid)]
   |                    ^^^^^^^

error: expected a power of 2 u32 literal
  --> tests/compile_fail/invalid_align_attr.rs:13:21
   |
13 |     #[shader_align(3)]
   |                     ^
